### PR TITLE
Update package versions to latest in nuget feed

### PR DESCRIPTION
**Pull request by calebkiage**
https://github.com/microsoft/kiota/issues/1439

### CODE REVIEW

File: src/msgraph-cli.csproj [hunk 1]
```diff
@@ -27,9 +27,8 @@
     <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="6.0.0" />
     <PackageReference Include="Microsoft.Extensions.Hosting" Version="6.0.1" />
     <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="6.0.0" />
-    <PackageReference Include="Microsoft.Kiota.Authentication.Azure" Version="1.0.8" />
-    <PackageReference Include="Microsoft.Kiota.Http.HttpClientLibrary" Version="1.0.22" />
-    <PackageReference Include="Microsoft.Kiota.Serialization.Json" Version="1.0.15" />
+    <PackageReference Include="Microsoft.Kiota.Authentication.Azure" Version="1.0.0-preview.1" />
```
Comment (andrueastman) [inline src/msgraph-cli.csproj:30]: Do you want to drop these as they are already defined in the project file of the `core` project?

If not, the `HttpClientLibrary` reference is missing

File: docker/init.sh [hunk 2]
```diff
@@ -2,9 +2,6 @@
 
 set -e
 
-script="$0"
-DIR="$(dirname $script)"
```
Comment (baywet) [inline docker/init.sh:6]: Was this removal intended here? If so can you provide more context please?

File: src/msgraph-cli.csproj (hunk 1 above)
Comment (baywet) [reply src/msgraph-cli.csproj:30]: Also missing serialization text

Comment (calebkiage) [reply src/msgraph-cli.csproj:30]: JSON and Azure aren't defined in the core

File: docker/init.sh (hunk 2 above)
Comment (calebkiage) [reply docker/init.sh:6]: Yes. The `$DIR` variable was created to find the directory the current script was running on. I then used it to run another script within the same directory in the created DBUS session. I forgot to remove it once I removed the additional script.

File: src/msgraph-cli.csproj (hunk 1 above)
Comment (andrueastman) [reply src/msgraph-cli.csproj:30]: They don't really need to be as they will still be pulled in by `Microsoft.Graph.Core` package defined in core. Yes?
The worry really is that if they are defined in different places, you may run into issues if (for example) `Microsoft.Graph.Core` requires a higher version than what is defined here. Thus, upgrading at `cli core` could cause issues here.

Comment (calebkiage) [reply src/msgraph-cli.csproj:30]: But that issue would only happen on updating the `Microsoft.Graph.Core`. In which case all the libraries would be updated? I can remove the dependencies, but any code issues caused by an incompatible update would need a keen eye to troubleshoot since these dependencies are defined in a package 2 levels deep in the dependency tree.

Comment (calebkiage) [reply src/msgraph-cli.csproj:30]: But that issue would only happen on updating the `Microsoft.Graph.Core`. In which case all the libraries would be updated? I can remove the dependencies, but any code issues caused by an incompatible update would need a keen eye to troubleshoot since these dependencies are defined in a package 2 levels deep in the dependency tree.

Comment (andrueastman) [reply src/msgraph-cli.csproj:30]: I don't think you can have a successful build if a package at `cli-core` defines a higher version that what is defined here. As that would violate the minimum requirements for the package at the higher level causing the build to fail. What you can do is get a package that is higher than the minimum requirements.

I believe we are saying the same thing. Essentially since you are bound to the minimum requirements of `graph core` the only reason to add the dependency is if you want a higher version of the package. If graph core updates a dependency you don't wish to have, you will still have to downgrade the `graph core` version to mitigate and not specify a lower version here. 

This comment applies to the `Microsoft.Kiota.Authentication.Azure` package as well.

Comment (calebkiage) [reply src/msgraph-cli.csproj:30]: I believe a person maintaining a project should be aware of transitive dependency upgrades of packages they directly use in case they need to update their code. If we remove the package dependencies, the build would pass but then some change in behavior could cause an issue later on.

**Review by andrueastman — APPROVED** [review]
